### PR TITLE
Make errors constants

### PR DESCRIPTION
**Pull request by caglar10ur**
since we don't want users to set them to nil

Signed-off-by: S.Çağlar Onur <caglar@10ur.org>

### CODE REVIEW

File: error.go [hunk 1]
```diff
@@ -6,82 +6,72 @@
...
-func NewError(msg string) error {
-	return &Error{
-		Message: msg,
-	}
-}
+type lxcError string
 
-func (e *Error) Error() string {
-	return e.Message
+func (e lxcError) Error() string {
```
Comment (brauner) [inline error.go:75]: Hm, isn't that an API break?

**Comment by tomponline** [comment]
@caglar10ur is this implementing this approach? https://dave.cheney.net/2016/04/07/constant-errors

File: error.go (hunk 1 above)
Comment (caglar10ur) [reply error.go:75]: In theory this should not break API (famous last words) All the checks like `if err == ErrAllocationFailed` or switch statements etc. should work as before.

**Comment by caglar10ur** [comment]
@tomponline yes, you are correct

**Comment by tomponline** [comment]
@caglar10ur saw this article the other night about a new package in go 1.13 called xerrors, looks interesting https://crawshaw.io/blog/xerrors

**Comment by caglar10ur** [comment]
@tomponline yep, I'm playing with it in $WORK - it is  good but I think we can wait a bit to see what will happen to it since I think it is still an active debate topic :)

**Pull request merged**